validator/client/src/test-support/java/tech/pegasys/artemis/validator/client/AttestationGenerator.java

```diff
@@ -0,0 +1,123 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.validator.client;
+
+import static tech.pegasys.artemis.datastructures.util.BeaconStateUtil.compute_epoch_of_slot;
+import static tech.pegasys.artemis.datastructures.util.BeaconStateUtil.get_domain;
+import static tech.pegasys.artemis.util.config.Constants.DOMAIN_ATTESTATION;
+import static tech.pegasys.artemis.util.config.Constants.MAX_VALIDATORS_PER_COMMITTEE;
+
+import com.google.common.primitives.UnsignedLong;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import org.apache.commons.lang3.tuple.MutableTriple;
+import org.apache.commons.lang3.tuple.Triple;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
+import tech.pegasys.artemis.datastructures.operations.Attestation;
+import tech.pegasys.artemis.datastructures.operations.AttestationData;
+import tech.pegasys.artemis.datastructures.state.BeaconState;
+import tech.pegasys.artemis.datastructures.state.CrosslinkCommittee;
+import tech.pegasys.artemis.datastructures.util.AttestationUtil;
+import tech.pegasys.artemis.storage.ChainStorageClient;
+import tech.pegasys.artemis.util.SSZTypes.Bitlist;
+import tech.pegasys.artemis.util.bls.BLSKeyPair;
+import tech.pegasys.artemis.util.bls.BLSPublicKey;
+import tech.pegasys.artemis.util.bls.BLSSignature;
+
+public class AttestationGenerator {
+  private final List<BLSKeyPair> validatorKeys;
+  private final BLSKeyPair randomKeyPair = BLSKeyPair.random();
+
+  public AttestationGenerator(final List<BLSKeyPair> validatorKeys) {
+    this.validatorKeys = validatorKeys;
+  }
+
+  public Attestation validAttestation(final ChainStorageClient storageClient) {
+    final Bytes32 bestBlockRoot = storageClient.getBestBlockRoot();
+    BeaconBlock block = storageClient.getStore().getBlock(bestBlockRoot);
+    BeaconState state = storageClient.getStore().getBlockState(bestBlockRoot);
+    return createAttestation(block, state, true);
+  }
+
+  public Attestation attestationWithInvalidSignature(final ChainStorageClient storageClient) {
+    final Bytes32 bestBlockRoot = storageClient.getBestBlockRoot();
+    BeaconBlock block = storageClient.getStore().getBlock(bestBlockRoot);
+    BeaconState state = storageClient.getStore().getBlockState(bestBlockRoot);
+    return createAttestation(block, state, false);
+  }
+
+  private Attestation createAttestation(
```

Comment:
I suspect we may be able to do a better job of re-using code here, but I didn't dig into it since we're in the middle of the 0.9 update.